clients/drcachesim/CMakeLists.txt

```diff
@@ -165,13 +165,15 @@ add_exported_library(drmemtrace_opcode_mix STATIC tools/opcode_mix.cpp)
 add_exported_library(drmemtrace_syscall_mix STATIC tools/syscall_mix.cpp)
 add_exported_library(drmemtrace_view STATIC tools/view.cpp)
```

Comment:
Please add a unit test similar to clients/drcachesim/tests/view_test.cpp.

Also some test that verifies the output for some checked in raw trace (e.g. clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/raw) like clients/drcachesim/tests/offline-view.templatex.

Reply:
Yes, that's in the TODO (see PR description).
However, before working on that, I'd like to come to a consensus on the pretty printed output of internal_record_view (<--- new tool name), as that is what I will test against.

Here's what it prints now (a short "representative" snippet, note that I OMITTED a few things because I wasn't sure I could share them here... the actual output has those values):
```
The internal_record_view tool is a pretty printer for trace_entry_t records.
The trace_entry_t internal record format is not a public interface and is subject to change without notice.
This tool is meant for debugging only, please do not rely on the details here.
<header, trace_version: 6 == frequent_timestamps>
<marker: version 6 frequent_timestamps>
<marker: filetype 0xe40 x86_64>
<thread, tid: 259013>
<pid, pid: 259013>
<marker: cache line size OMITTED>
<marker: chunk instruction count 10000000>
<marker: page size OMITTED>
<marker: timestamp 13352585958813850>
<marker: cpu id 3>
<encoding, num_encoding_bytes: 8, encoding_bytes: OMITTED>
<instr, length: 8, pc: 0x7f6fdd3ec360>
<encoding, num_encoding_bytes: 8, encoding_bytes: OMITTED>
<direct_call, length: 8, pc: 0x7f6fdd3ec368>
<write, memref_size: 8, memref_addr: 0x7fffa4ef43c8>
<encoding, num_encoding_bytes: 8, encoding_bytes: OMITTED>
<instr, length: 8, pc: 0x7f6fdd3ec370>
<write, memref_size: 8, memref_addr: 0x7fffa4ef43c0>
...
```

Reply:
I don't think it's worth spending any more time on this (esp for things like dividers and ordinals that the main view tool already has): this PR is just adding an internal tool that honestly will likely be rarely used, as evidenced by it not being needed in the past. Even now there likely is no use case where you wouldn't also go run the final view tool afterward, since that's the real view seen by tools.  E.g., for your own ISA transform and filtering, you can't just go by the output of this record view tool: you have to make sure it works at the memref_t layer, so you have to use the real view tool plus invariant checks anyway.  This internal tool is only a limited-use debugging step pretty much always followed up by using the "real" pretty printer.

Reply:
Thinking more about when I've looked at the on-disk view with `od`: the most recent was debugging record_filter zip chunking support.  There I had to look just before and after the 10M-instr chunk boundaries.  Running `od` on one particular zipfile subfile at a time made this very easy.  Looking at the output above with no ordinals, and having to pass in the whole trace file/dir at once: how would I focus on chunk boundaries? I might end up going back to `od`!  I guess `-skip_instructions` is technically implemented for record_reader_t?  But it's slow: the fast chunk skips are not implemented.  Plus if the record_filter being debugged got the chunk boundary wrong I would have to go search for the chunk ord marker; if that's missing or wrong: would have to fall back to `od`.

Not sure there's an action item here: just an observation that this new tool is not enough to debug all on-disk issues as it abstracts away the file splits and I'm not sure it would have been useful the last time I did that.  Adding ordinals would help -- but I don't know it's worth the effort.  I would probably go run the regular view tool skipped to the chunk transition and take the closest timestamps and search for that in this tool's output or sthg.  But I would again want to be at the real file layer instead of the trace_entry_t stream layer's abstracting away of the file divisions.